### PR TITLE
Documentation for pgAdmin

### PR DESCRIPTION
**Pull request by jmckulk**
This commit adds documentation to deploy pgAdmin with a Postgres
cluster. It includes instructions for deployment and deletion and notes
on user synchronization.

**Checklist:**

 
 - [ ] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [ ] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**

 
 - [ ] New feature
 - [ ] Bug fix
 - [x] Documentation
 - [ ] Testing enhancement
 - [ ] Other

**What is the current behavior (link to any open issues here)?**

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

**Other Information**:
[sc-13238]

### CODE REVIEW

File: docs/content/architecture/pgadmin4.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: "pgAdmin 4"
+date:
+draft: false
+weight: 900
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:5]: Ha, how did you come up with this weight? "We may add other stuff, but I want pgAdmin to always be last"?

(Seriously, I was going to suggest going lighter than 900 since the next biggest is, what, 140; but since we can change this weight on any deploy, if we decide we need to move it or make space, we can do that then.)

File: docs/content/architecture/pgadmin4.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Let's use the PostgreSQL cluster we created in the [quickstart]({{< relref "quickstart/_index.md" >}})
+as an example. If you don't have the `hippo` example cluster running, follow those steps.
+
+After the PostgreSQL cluster becomes ready, you can create a pgAdmin 4 deployment. To add the
+deployment to your cluster, you must add the following fields to your spec. 
+
+```
+  userInterface:
+    pgAdmin:
+      image: registry.developers.crunchydata.com/crunchydata/crunchy-pgadmin4:centos8-13.5-4.5.5
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:32]: ⛏️ Would this be easier to maintain with an constant in docs/config.toml and the `{{< param imageCrunchyPgAdmin >}}` format?

File: docs/content/architecture/pgadmin4.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      dataVolumeClaimSpec:
+        accessModes:
+        - "ReadWriteOnce"
+        resources:
+          requests:
+            storage: 1Gi
+```
+
+This creates a pgAdmin 4 deployment unique to this PostgreSQL cluster and synchronizes the
+PostgreSQL user information. To access pgAdmin 4, you can set up a port-forward to the Service,
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:42]: ⛏️ Why "Service" rather than "service" -- is that a pattern we have?

File: docs/content/architecture/pgadmin4.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+## User Synchronization
+
+The operator will synchronize users defined in the spec (e.g., in `spec.users`) with the pgAdmin 4
+deployment. Any user created in the database without being defined in the spec will not be
+synchronized.
+
+## Deleting pgAdmin 4
+
+You can remove the pgAdmin 4 deployment by removing the `userInterface` spec field.
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:80]:
```suggestion
You can remove the pgAdmin 4 deployment by removing the `userInterface` field from the spec.
```

File: docs/content/architecture/pgadmin4.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+title: "pgAdmin 4"
+date:
+draft: false
+weight: 900
+---
+
+![pgAdmin 4 Query](/images/pgadmin4-query.png)
+
+[pgAdmin 4](https://www.pgadmin.org/) is a popular graphical user interface that
+makes it easy to work with PostgreSQL databases from both a desktop or web-based
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:11]: ❓ Since we mention the desktop client, is it worth showing how to set up that (I don't think so) or link to the pgAdmin docs on that (maybe)?

File: docs/content/architecture/pgadmin4.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+PostgreSQL cluster and keeps users' database credentials synchronized. You can
+simply log into pgAdmin 4 with your PostgreSQL username and password and
+immediately have access to your databases.
+
+## Deploying pgAdmin 4
+
+Let's use the PostgreSQL cluster we created in the [quickstart]({{< relref "quickstart/_index.md" >}})
+as an example. If you don't have the `hippo` example cluster running, follow those steps.
+
+After the PostgreSQL cluster becomes ready, you can create a pgAdmin 4 deployment. To add the
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:26]: Just to double-check, what happens if we try to add the `userInterface` field the first time we create a postgres cluster?

**Review by benjaminjb — COMMENTED** [review]
LG, just a few questions

File: docs/content/architecture/pgadmin4.md (hunk 1 above)
Comment (jmckulk) [reply docs/content/architecture/pgadmin4.md:5]: Haha, I did not. That was the weight from the old docs. We can for sure make it lower

File: docs/content/architecture/pgadmin4.md (hunk 3 above)
Comment (jmckulk) [reply docs/content/architecture/pgadmin4.md:42]: I would assume because it is a Kubernetes object, but again I copied from the old docs.

Comment (benjaminjb) [reply docs/content/architecture/pgadmin4.md:42]: You know, I just double-checked, and it looks like we do have that pattern of capitalizing Service.

Comment (benjaminjb) [reply docs/content/architecture/pgadmin4.md:42]: Except in docs/content/tutorial/connect-cluster.md line 52, where it appears lowercased.

Is that worth changing?

File: docs/content/architecture/pgadmin4.md (hunk 5 above)
Comment (jmckulk) [reply docs/content/architecture/pgadmin4.md:11]: I could look something up and add it. We didn't have anything in the old docs, so it might not be worth it.

File: docs/content/architecture/pgadmin4.md (hunk 6 above)
Comment (jmckulk) [reply docs/content/architecture/pgadmin4.md:26]: It does come up as expected. 

I could also update the docs to have it be one step :thinking: . Should I provide one spec with the UI fields already added :question: I was following steps from the old docs

Comment (benjaminjb) [reply docs/content/architecture/pgadmin4.md:26]: I think it makes sense to continue this as if from the quickstart (with everything that implies, e.g., a hippo user), but I'd maybe reword this section a bit in that case, something like "if you've done the quickstart, add this section to the spec and reapply; if you don't have any postgres cluster running, add this section to a spec with a user specified, and apply." (But, you know, better said.)

I just think if I came to this cold and read "After the PostgreSQL cluster becomes ready, you can create a pgAdmin 4 deployment", I might think it was a two-step process.

File: docs/content/architecture/pgadmin4.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
+---
+title: "pgAdmin 4"
```
Comment (cbandy) [inline docs/content/architecture/pgadmin4.md:2]: 🤔 I can't tell if the upstream project is called "pgAdmin" or "pgAdmin 4"... The website and docs seem to use both all over. Do we want this URL to be `.../pgadmin4` long-term, or should we hedge and have it be `.../pgadmin`?

Either way, I'm fine with the title being pgAdmin 4.

https://www.pgadmin.org/faq/
> **What is pgAdmin 4?**
> pgAdmin is …
>
> **What browsers does pgAdmin support?**
> The following browsers are supported with pgAdmin…

File: docs/content/architecture/pgadmin4.md (hunk 5 above)
Comment (cbandy) [reply docs/content/architecture/pgadmin4.md:11]: 🤔 In the context of this paragraph, I'm inclined to drop the "desktop" mention. The very next sentence says the operator will help with pgAdmin 4, but that's not the case for a desktop install/deployment.

File: docs/content/architecture/pgadmin4.md (hunk 7 above)
Comment (jmckulk) [reply docs/content/architecture/pgadmin4.md:2]: If `pgadmin` works, I would prefer that, but I don't have any strong feelings either way.

File: docs/content/architecture/pgadmin4.md (hunk 5 above)
Comment (jmckulk) [reply docs/content/architecture/pgadmin4.md:11]: Oh yeah, that is weird. The comment is saying pgAdmin works with the desktop app (and I assume it does), but that has nothing to do with us.

**Review by tjmoore4 — COMMENTED** [review]
:+1: Looks good overall, just one change regarding the desktop client and some clarification on when pgAdmin can be added.

**Review by tjmoore4 — APPROVED** [review]
With updates, looks good to me :+1:

File: docs/content/architecture/pgadmin4.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+The PostgreSQL Operator lets you deploy a pgAdmin 4 environment alongside a
+PostgreSQL cluster and keeps users' database credentials synchronized. You can
+simply log into pgAdmin 4 with your PostgreSQL username and password and
+immediately have access to your databases.
+
+## Deploying pgAdmin 4
+
+If you've done the [quickstart]({{< relref "quickstart/_index.md" >}}), add the
+following fields to the spec and reapply; If you don't have any Postgres clusters
```
Comment (benjaminjb) [inline docs/content/architecture/pgadmin4.md:24]:
```suggestion
following fields to the spec and reapply; if you don't have any Postgres clusters
```

**Review by benjaminjb — APPROVED** [review]
